application_sdk/outputs/eventstore.py

```diff
@@ -146,32 +69,42 @@ def enrich_event_metadata(cls, event: Event):
         return event
 
     @classmethod
-    def publish_event(cls, event: Event, enrich_metadata: bool = True):
-        """Create a new generic event.
+    async def publish_event(cls, event: Event):
+        """Publish event with automatic fallback between pub/sub and HTTP binding.
 
         Args:
-            event (Event): Event data.
-            topic_name (str, optional): Topic name to publish the event to. Defaults to TOPIC_NAME.
-
-        Example:
-            >>> EventStore.create_generic_event(Event(event_type="test", data={"test": "test"}))
+            event (Event): Event data to publish.
         """
         if not is_component_registered(EVENT_STORE_NAME):
             logger.warning(
                 "Skipping event publish because event store component is not registered",
             )
             return
         try:
-            if enrich_metadata:
-                event = cls.enrich_event_metadata(event)
+            event = cls.enrich_event_metadata(event)
+
+            payload = json.dumps(event.model_dump(mode="json"))
+
+            # Prepare binding metadata with auth token for HTTP bindings
+            binding_metadata = {"content-type": "application/json"}
+
+            # Add auth token - HTTP bindings will use it, others will ignore it
+            auth_client = AtlanAuthClient()
+            auth_token = await auth_client.get_access_token()
+            binding_metadata["Authorization"] = f"Bearer {auth_token}"
```

Comment:
1. this block might be reused in the future
2. we have a method `get_authenticated_headers()` in `atlan_auth.py` which is currently unused 

could you just move this block to `get_authenticated_headers` method and use that to return the dict that is needed?